### PR TITLE
Runtime core properties

### PR DESCRIPTION
**Pull request by junmendoza**
@ke-yu - PTAL

Moved from core to runtimecore
1. ExecMode
2. InterpreterProperties
3. ReplicationGuide cache

### CODE REVIEW

File: src/Engine/ProtoAssociative/CodeGen.cs [hunk 1]
```diff
@@ -3957,7 +3957,7 @@ public override int Emit(ProtoCore.AST.Node codeBlockNode, ProtoCore.Associative
             AllocateContextGlobals();
 
             core.startPC = this.pc;
-            if (core.ExecMode == ProtoCore.DSASM.InterpreterMode.kExpressionInterpreter)
+            if (core.Options.RunMode == ProtoCore.DSASM.InterpreterMode.kExpressionInterpreter)
```
Comment (ke-yu) [inline src/Engine/ProtoAssociative/CodeGen.cs]: A question, why compiler core needs this ``RunMode``? Seems, well, ``RunMode`` is runtime stuff.

File: test/Engine/ProtoTestFx/WatchTestFx.cs [hunk 2]
```diff
@@ -140,7 +140,7 @@ protected override void POP_Handler(Instruction instruction)
             {
                 int lineNo = instruction.debug.Location.StartInclusive.LineNo;
 
-                if (Core.Options.IDEDebugMode && Core.ExecMode != ProtoCore.DSASM.InterpreterMode.kExpressionInterpreter)
```
Comment (ke-yu) [inline test/Engine/ProtoTestFx/WatchTestFx.cs]: Why ``IDEDebugMode`` is an enumeration value of ``Options.RunMode``?

**Comment by ke-yu** [comment]
Overall it is LGTM except these two minor questions.

File: test/Engine/ProtoTestFx/WatchTestFx.cs (hunk 2 above)
Comment (junmendoza) [inline test/Engine/ProtoTestFx/WatchTestFx.cs]: Yea it should have been
I will move this in http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6193
This is part of this sprint

File: src/Engine/ProtoAssociative/CodeGen.cs (hunk 1 above)
Comment (junmendoza) [inline src/Engine/ProtoAssociative/CodeGen.cs]: Im still trying to determine is runmode needs its own Options. Right now, the Options passed to the runtime is the same as the Option used by the compile core.

Added this investigation to: http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6193
Targetted for this sprint

**Pull request merged**